plugin/src/main/java/com/denizenscript/denizen/objects/PlayerTag.java

```diff
@@ -104,13 +104,20 @@ public static PlayerTag mirrorBukkitPlayer(OfflinePlayer player) {
      * Notes that the player exists, for easy PlayerTag valueOf handling.
      */
     public static void notePlayer(OfflinePlayer player) {
-        if (player.getName() == null) {
-            Debug.echoError("Null named player " + player + " - may be file corruption, or player data imported from non-bukkit server?");
+        UUID uuid = player.getUniqueId();
+        notePlayer(NMSHandler.playerHelper.getOfflineData(uuid).getName(), uuid);
+    }
+
+    public static void notePlayer(Player player) {

```

Comment:
This is wrong -- OfflinePlayer, does not, uh, mean a player that is offline. It means a player that *could be* offline. You need `isOnline()` to differentiate whether an OfflinePlayer is online or not.